src/gui/editors/SongEditor.cpp

```diff
@@ -523,6 +523,15 @@ void SongEditor::keyPressEvent( QKeyEvent * ke )
 	{
 		selectAllTcos( false );
 	}
+	else if ( ke->key() == Qt::Key_Space && ke->modifiers() & Qt::ShiftModifier )
+	{
+		tick_t t = m_song->currentTick();
+		if ( t >= 0 && m_song->isPlaying() )
+		{
+			m_song->stop();
+			m_song->setPlayPos( t, Song::Mode_PlaySong );
+		}
+	}
```

Comment:
I think this code is okay. I don't have write access either ways, but I think you should rebase to fix all those other files that are messed up in the PR, and I think you'll be ready for a merge (which I can't do).